doc/stdenv/platform-notes.chapter.md

```diff
@@ -60,3 +60,34 @@ Some common issues when packaging software for Darwin:
   ```
 
   The package `xcbuild` can be used to build projects that really depend on Xcode. However, this replacement is not 100% compatible with Xcode and can occasionally cause issues.
+
+- x86_64-darwin uses the 10.12 SDK by default, which is built from the Apple open source releases. For software that is not
+  compatible with the 10.12 SDK, the 11.0 SDK used by aarch64-darwin can be used. It is available under the `apple_sdk_11_0` attr
+  and requires some extra steps to be taken to use it. Both `clang` and `stdenv` will need to be overriden. See below for an example.
+
+  ```nix
+  { stdenv, overrideCC, xcbuild, AppKit, Foundation, Libsystem }:
+  let
+    clang_11_0 = stdenv.cc.override {
+      bintools = stdenv.cc.bintools.override { libc = Libsystem; };
+      libc = Libsystem;
+    };
+    stdenv_11_0 =
+      if stdenv.isx86_64 then
+        (overrideCC stdenv clang_11_0).override
+          {
+            targetPlatform = stdenv.targetPlatform // {
+              darwinMinVersion = "10.12";
+              darwinSdkVersion = "11.0";
+            };
+          }
+      else
+        stdenv;
+      xcbuild_11_0 = xcbuild.override { stdenv = stdenv_11_0; };
+    in
+    stdenv_11_0.mkDerivation {
+      buildInputs = [ AppKit Foundation ];
+      nativeBuildInputs = [ xcbuild_11_0 ];
+      # ...
+    }
```

Comment:
If we want to use this in nixpkgs I would suggest the following points.
- be very careful. I think mixing SDKs could easily backfire and we need to carefully test this to make sure we don't break everything.
- Can we add this to top level to make sure all packages use the same overrides? It would also make updating easier in case we discover bugs and need to change things later.

Reply:
After discussing it with @toonn on Matrix, I pushed an update that adds a `stdenv` attribute to the 11.0 SDK attribute.